presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt

```diff
@@ -1712,6 +1714,86 @@ class HomeScreenViewModel(
         }
     }
 
+    fun onCreateObjectForWidget(
+        type: ObjectWrapper.Type,
+        widget: Id,
+        source: Id
+    ) {
+        viewModelScope.launch {
+            createObject.async(
+                params = CreateObject.Param(
+                    type = type.uniqueKey?.let {
+                        TypeKey(it)
+                    }
+                )
+            ).fold(
+                onSuccess = { result ->
+                    proceedWithCreatingLinkToNewObject(source, result)
+                    proceedWithNavigation(result.obj.navigation())
+                },
+                onFailure = {
+                    Timber.e(it, "Error while creating object")
+                }
+            )
+        }
+    }
+
+    private suspend fun proceedWithCreatingLinkToNewObject(
+        source: Id,
+        result: CreateObject.Result
+    ) {
+        createBlock.async(
+            params = CreateBlock.Params(
+                context = source,
+                target = "",
+                position = Position.NONE,
+                prototype = Block.Prototype.Link(
+                    target = result.objectId
+                )
+            )
+        ).fold(
+            onSuccess = {
+
```

Comment:
why it's empty here?

Reply:
added comment. we do not need to handle request's response here.